text/0000-return-position-impl-trait-in-traits.md

```diff
@@ -0,0 +1,508 @@
+---
+tags: draft-rfc
+title: Return position `impl Trait` in traits
+---
+
+- Feature Name: return_position_impl_trait_in_traits
+- Start Date: 2023-04-27
+- RFC PR: [rust-lang/rfcs#3193](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+- Initiative: [impl trait initiative](https://github.com/rust-lang/impl-trait-initiative)
+
+# Summary
+[summary]: #summary
+
+* Permit `impl Trait` in fn return position within traits and trait impls.
+* Allow `async fn` in traits and trait impls to be used interchangeably with its equivalent `impl Trait` desugaring.
```

Comment:
One consequence of this is that it basically stabilizes the `impl Future` desugaring. I don't imagine we'll want to change it, and enough things depend on the `impl Future` desugaring now that we probably can't change it practically already, but it still means we're committing to what in some ways feel like internal implementation details to me.